### PR TITLE
Improved exception logging

### PR DESCRIPTION
**Pull request by isaac-s**

### CODE REVIEW

**Review by mabuaisha — APPROVED** [review]

File: cloudify_kubernetes/tests/test_tasks.py [hunk 1]
```diff
@@ -866,9 +866,11 @@ def to_dict(self):
                                 resource_definition=None
                             )
                         file_mock.assert_called_with('new_path')
-
+        print(error.exception)
+        import json
+        print(json.dumps(error.exception.causes, indent=4))
```
Comment (EarthmanT) [inline cloudify_kubernetes/tests/test_tasks.py:871]: Shall we remove this stuff?

File: cloudify_kubernetes/decorators.py [hunk 2]
```diff
@@ -32,7 +31,6 @@
 from .k8s import (CloudifyKubernetesClient,
                   KubernetesApiAuthenticationVariants,
                   KubernetesApiConfigurationVariants,
-                  KuberentesApiInitializationFailedError,
```
Comment (EarthmanT) [inline cloudify_kubernetes/decorators.py:35]: Is this not possible?

File: cloudify_kubernetes/tests/test_tasks.py (hunk 1 above)
Comment (isaac-s) [reply cloudify_kubernetes/tests/test_tasks.py:871]: Damn

File: cloudify_kubernetes/decorators.py (hunk 2 above)
Comment (isaac-s) [reply cloudify_kubernetes/decorators.py:35]: Is is, but it doesn't require any special handling rather than the one we do for `RecoverableException`

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**